src/domain/event/EventOccurrenceNotificationControlButton.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+
+import Button from '../../common/components/button/Button';
+import EventNotificationSubscriptionModal, {
+  Occurrence,
+} from './EventNotificationSubscriptionModal';
+import useUnsubscribeFromFreeSpotNotificationMutation from './useUnsubscribeFromFreeSpotNotificationMutation';
+import styles from './eventOccurrenceNotificationControlButton.module.css';
+
+type Props = {
+  childId: string;
+  eventId: string;
+  isSubscribed: boolean;
+  occurrence: Occurrence;
+  onUnsubscribed?: () => void;
+  onSubscribe?: () => void;
+  onSubscribed?: () => void;
+  subscribeLabel: string;
+  unsubscribeLabel: string;
+};
+
+const EventNotificationControlButton = ({
+  childId,
+  eventId,
+  isSubscribed,
+  occurrence,
+  onUnsubscribed,
+  onSubscribe,
+  onSubscribed,
+  subscribeLabel,
+  unsubscribeLabel,
+}: Props) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [unsubscribe] = useUnsubscribeFromFreeSpotNotificationMutation(
```

Comment:
I don't remember how strict we have been about adding types to `useState` hooks.

Reply:
Me neither, but I think that it makes sense to add one here, thanks for pointing this out 👍